test/unit/StakingPool/calculatePremium.js

```diff
@@ -0,0 +1,611 @@
+const { expect } = require('chai');
+const { ethers } = require('hardhat');
+const { parseEther } = ethers.utils;
+const { BigNumber } = ethers;
+const {
+  calculateSurgePremiums,
+  calculateBasePrice,
+  divCeil,
+  calculateFixedPricePremium,
+  calculatePriceBump,
+} = require('./helpers');
+const { daysToSeconds } = require('../../../lib/helpers');
+
+describe('calculatePremium', function () {
+  // If an arithmetic operation results in underflow or overflow outside of an unchecked { ... } block.
+  const UNDER_OR_OVERFLOW = 0x11;
+  // If you divide or modulo by zero (e.g. 5 / 0 or 23 % 0).
+  const DIVIDE_BY_ZERO = 0x12;
+
+  const stakedProductTemplate = {
+    lastEffectiveWeight: BigNumber.from(50),
+    targetWeight: BigNumber.from(70), // 70%
+    targetPrice: BigNumber.from(200), // 2%
+    nextPrice: BigNumber.from(200), // 2%
+    nextPriceUpdateTime: BigNumber.from(0),
+  };
+
+  it('should calculate premium on multiple cover buys over time, based on pre-defined numbers', async function () {
+    const { stakingPool } = this;
+    let { timestamp } = await ethers.provider.getBlock('latest');
+    timestamp = BigNumber.from(timestamp);
+    const { NXM_PER_ALLOCATION_UNIT, PRICE_BUMP_RATIO, PRICE_CHANGE_PER_DAY, INITIAL_PRICE_DENOMINATOR } = this.config;
+    let stakedProduct = { ...stakedProductTemplate, nextPriceUpdateTime: timestamp };
+    const period = daysToSeconds(1);
+    const totalCapacity = divCeil(parseEther('50000'), NXM_PER_ALLOCATION_UNIT);
+
+    // 1st cover buy
+    {
+      const coverAmountRaw = parseEther('2400');
+      const coverAmount = divCeil(coverAmountRaw, NXM_PER_ALLOCATION_UNIT);
+      const initialCapacityUsed = BigNumber.from(0);
+      const basePrice = calculateBasePrice(timestamp, stakedProduct, PRICE_CHANGE_PER_DAY);
+      expect(basePrice).to.be.equal(200);
+      const priceBump = calculatePriceBump(coverAmount, PRICE_BUMP_RATIO, totalCapacity);
+      expect(priceBump).to.be.equal(96);
```

Comment:
i understand why the javascript functions were introduced - to kind of show how values were derived

The issue as i think @shark0der  mentioned is that you can have the same bug in solidity and in js and gloss over

For this test it's matching the values shown by Hugh in the sheet, we can hardcode the values here as i can see you partially did with these extra assertions


From a test writing point of view, it's not exactly a test to assert the result of a javascript function that's used for testing. you are testing the testing code at that point

I can see this test written with hardcoded values from Hugh's sheet, instead of of the testing function